include/ck_tile/ops/gemm/pipeline/gemm_pipeline_agmem_bgmem_creg_v1_default_policy.hpp

```diff
@@ -486,30 +494,125 @@ struct GemmPipelineAGmemBGmemCRegV1DefaultPolicy
-                                                                      typename Problem::BDataType,
-                                                                      typename Problem::CDataType,
-                                                                      BlockWarps,
-                                                                      WarpGemm>;
-
-        return BlockUniversalGemmAsBsCr<Problem, BlockGemmPolicy>{};
+        if constexpr(Problem::kBlockDefaultPolicy == true)
+        {
+            // Default Policy
+            if constexpr(Problem::kBlockPolicyMethod == 0)
+            {
+                using BlockGemmPolicy = BlockGemmARegBRegCRegV1DefaultPolicy;
+                return BlockGemmARegBRegCRegV1<Problem, BlockGemmPolicy>{};
+            }
+            else if constexpr(Problem::kBlockPolicyMethod == 1)
+            {
+                using BlockGemmPolicy = BlockGemmARegBRegCRegV2DefaultPolicy;
+                return BlockGemmARegBRegCRegV1<Problem, BlockGemmPolicy>{};
+            }
+        }
+        else
+        {
+            // Custom Policy
+            constexpr bool TransposeC = false;
+            constexpr auto I0         = number<0>{};
+            constexpr auto I1         = number<1>{};
+            constexpr auto I2         = number<2>{};
+
+            using AccDataType     = float;
+            using BlockWarps      = typename Problem::BlockGemmShape::BlockWarps;
+            using WarpTile        = typename Problem::BlockGemmShape::WarpTile;
+            using WarpGemm        = WarpGemmMfmaDispatcher<typename Problem::ADataType,
+                                                    typename Problem::BDataType,
+                                                    AccDataType,
+                                                    WarpTile::at(I0),
+                                                    WarpTile::at(I1),
+                                                    WarpTile::at(I2),
+                                                    TransposeC>;
+            using BlockGemmPolicy = BlockGemmARegBRegCRegV1CustomPolicy<typename Problem::ADataType,
+                                                                        typename Problem::BDataType,
+                                                                        typename Problem::CDataType,
+                                                                        BlockWarps,
+                                                                        WarpGemm>;
+
+            if constexpr(Problem::kBlockPolicyMethod == 2)
+            {
+                return BlockGemmARegBRegCRegV1<Problem, BlockGemmPolicy>{};
+            }
+            else
+            {
+                static_assert(
+                    false,
+                    "There are no other customer policies to support in BlockUniversalGemmAsBsCr");
+            }
+        }
```

Comment:
All those block gemm, and block gemm pipeline selection logic is really bad. We can't hardfix this like that. It would be totally unmaintainable. Instead of that it would be much cleaner to resort to sth similar as in block gemms, that is custom policies which are parameterized with ie. block gemm & block gemm policy user would like to use.